src/vs/workbench/parts/terminal/node/terminalPanel.ts

```diff
@@ -0,0 +1,145 @@
+
+	private ptyProcess: Terminal;
+	private parentDomElement: HTMLElement;
+	private terminal;
+	private terminalDomElement: HTMLDivElement;
+
+	constructor(
+		@IConfigurationService private configurationService: IConfigurationService,
+		@ITelemetryService telemetryService: ITelemetryService,
+		@IWorkspaceContextService private contextService: IWorkspaceContextService
+	) {
+		super(TERMINAL_PANEL_ID, telemetryService);
+	}
+
+	public layout(dimension: Dimension): void {
+		let cols = Math.floor(this.parentDomElement.offsetWidth / TERMINAL_CHAR_WIDTH);
+		let rows = Math.floor(this.parentDomElement.offsetHeight / TERMINAL_CHAR_HEIGHT);
+		this.terminal.resize(cols, rows);
+		this.ptyProcess.resize(cols, rows);
+	}
+
+	public create(parent: Builder): TPromise<void> {
+		super.create(parent);
+
+		this.parentDomElement = parent.getHTMLElement();
+		this.createTerminal();
+
+		return TPromise.as(null);
+	}
+
+	private createTerminal(): void {
+		this.parentDomElement.innerHTML = '';
+		this.ptyProcess = fork(this.getShell(), [], {
+			name: fs.existsSync('/usr/share/terminfo/x/xterm-256color') ? 'xterm-256color' : 'xterm',
+			cols: 80,
+			rows: 6,
+			cwd: this.contextService.getWorkspace() ? this.contextService.getWorkspace().resource.path : process.env.HOME
+		});
+		this.terminalDomElement = document.createElement('div');
+		this.parentDomElement.classList.add('integrated-terminal');
+		let terminalScrollable = new DomNodeScrollable(this.terminalDomElement);
+		let terminalContainer = new ScrollableElement(this.terminalDomElement, terminalScrollable, { horizontal: 'hidden', vertical: 'auto' });
+		this.terminal = termJs({
+			cursorBlink: false // term.js' blinking cursor breaks selection
+		});
+
+		this.ptyProcess.on('data', (data) => {
+			this.terminal.write(data);
+		});
+		this.terminal.on('data', (data) => {
+			this.ptyProcess.write(data);
+			return false;
+		});
+		this.ptyProcess.on('exit', (data) => {
+			this.terminal.destroy();
+			// TODO: When multiple terminals are supported this should do something smarter. There is
+			// also a weird bug here at leasy on Ubuntu 15.10 where the new terminal text does not
+			// repaint correctly.
+			this.createTerminal();
+		});
+		this.parentDomElement.addEventListener('mousedown', (event) => {
+			// Drop selection and focus terminal on Linux to enable middle button paste when click
+			// occurs on the selection itself.
+			if (event.which === 2 && platform.isLinux) {
+				this.focusTerminal(true);
+			}
+		});
+		this.parentDomElement.addEventListener('mouseup', (event) => {
+			if (event.which !== 3) {
+				this.focusTerminal();
+			}
+		});
+
+		this.terminal.open(this.terminalDomElement);
+		this.parentDomElement.appendChild(terminalContainer.getDomNode());
+
+		let config = this.configurationService.getConfiguration<ITerminalConfiguration>();
+		this.terminalDomElement.style.fontFamily = config.integratedTerminal.fontFamily;
+		this.terminal.colors = this.getTerminalColors();
```

Comment:
unsure where/how that is configured but the terminal shows dark tho I use a light theme...